src/Nethermind/Nethermind.Network/Rlpx/NettyHandshakeHandler.cs

```diff
@@ -110,7 +110,15 @@ public override void ChannelRegistered(IChannelHandlerContext context)
 
         public override void ExceptionCaught(IChannelHandlerContext context, Exception exception)
         {
-            string clientId = _session?.Node?.ToString(Node.Format.Console) ?? $"unknown {_session?.RemoteHost}";
+            string clientId = $"unknown {_session?.RemoteHost}";
+            try
+            {
+                clientId = _session?.Node?.ToString(Node.Format.Console);
```

Comment:
why does this throw exceptions? Can we have a version that is guaranteed not to throw?

Reply:
The `Node` property throw exception because `RemoteId` is still missing at this point. Changing that to a nullable might make other place more inconviniecne.

Reply:
Can we handle that in `ToString` method and not throw exceptions?

Reply:
Which `ToString`? The `Node` property of the `_session` that is throwing.

Reply:
Cosmetic: instead of having empty catch we could move the below checks to session bool property: 

```
              if (_node is null)
                {
                    if (RemoteNodeId is null || RemoteHost is null || RemotePort == 0)
                    {
```

Reply:
Ah can we introduce Session.TryGetNode?

Reply:
I just add `if (_session.RemoteNodeId != null) ` in front of it.